src/transactionsFactories/tokenManagementTransactionsFactory.ts

```diff
@@ -575,14 +583,14 @@ export class TokenManagementTransactionsFactory {
     createTransactionForBurningQuantity(options: {
         sender: IAddress;
         tokenIdentifier: string;
-        tokenNonce: BigNumber.Value;
-        quantityToBurn: BigNumber.Value;
+        tokenNonce: bigint;
+        quantityToBurn: bigint;
     }): TransactionNext {
         const dataParts = [
             "ESDTNFTBurn",
             utf8ToHex(options.tokenIdentifier),
-            bigIntToHex(options.tokenNonce),
-            bigIntToHex(options.quantityToBurn),
+            bigIntToHex(new BigNumber(options.tokenNonce.toString())),
+            bigIntToHex(new BigNumber(options.quantityToBurn.toString())),
```

Comment:
Perhaps use `numberToPaddedHex` (which handles `bigint`). If necessary, maybe handle 0 differently in `numberToPaddedHex`.

Reply:
Replaced with `numberToPaddedHex`.